pandas/io/parquet.py

```diff
@@ -154,6 +155,32 @@ def __init__(self):
 
         self.api = pyarrow
 
+    @staticmethod
+    def _write_attrs(table, df: DataFrame):
+        schema_metadata = table.schema.metadata or {}
+        pandas_metadata = json.loads(schema_metadata.get(b"pandas", "{}"))
+        column_attrs = {}
+        for col in df.columns:
+            attrs = df[col].attrs
+            if not attrs or not isinstance(col, str):
+                continue
+            column_attrs[col] = attrs
+        pandas_metadata.update(
+            attrs=df.attrs,
+            column_attrs=column_attrs,
+        )
+        schema_metadata[b"pandas"] = json.dumps(pandas_metadata)
+        return table.replace_schema_metadata(schema_metadata)
+
+    @staticmethod
+    def _read_attrs(table, df: DataFrame):
+        schema_metadata = table.schema.metadata or {}
```

Comment:
make this a free function, type table and the output. add a doc-string

Reply:
I wasn't sure how to type `table` since pyarrow is an optional dependency. Do you have an example of how to handle this?

Reply:
I used `Any` for now.